packages/workbox-core/utils/cacheWrapper.mjs

```diff
@@ -0,0 +1,83 @@
+const putWrapper = async (request, response, plugins) => {
+  if (!plugins) {
+    plugins = [];
+  }
+
+  let responseToCache = await _isResponseSafeToCache(
+    request, response, plugins);
+  if (!responseToCache) {
+    return;
+  }
+
+  const CACHE_NAME = 'TODO-CHANGE-ME';
+  const cache = await caches.open(CACHE_NAME);
+
+  let matchOptions;
+  let oldResponse = await matchWrapper(request, matchOptions);
+
+  // Regardless of whether or not we'll end up invoking
+  // cacheDidUpdate, wait until the cache is updated.
+  await cache.put(request, responseToCache);
+
+  for (let plugin of plugins) {
+    if (plugin.cacheDidUpdate) {
+      await plugin.cacheDidUpdate(CACHE_NAME,
+        request,
+        oldResponse,
+        responseToCache,
+      );
+    }
+  }
+
+  /**
+
+  // Only conditionally await the caching completion, giving developers the
+  // option of returning early for, e.g., read-through-caching scenarios.
+  if (waitOnCache) {
```

Comment:
cacheWrapper.put will always wait on the operation. The main use for it was so that precaching could block on the cache put when fetchAndCache originally didn't.

With these being seperated, it's up to the implementation to block or not on the operation.